luncheon/headers.lua

```diff
@@ -135,11 +135,13 @@ end
 
 ---Insert a single key value pair to the collection will duplicate existing keys
 ---@param key string
----@param value string
+---@param value string|nil
 ---@return Headers
 function Headers:append(key, value)
   key = Headers.normalize_key(key)
-  _append(self._inner, key, value)
+  -- because parsing a value-less header line populates the map with an empty string,
+  -- we normalize passed nil values to empty string.
+  _append(self._inner, key, (value or ''))
```

Comment:
I wonder if the call site should be doing the `or` here?

Reply:
I thought about that, but I wanted to provide the option for callers to be able to use `header:append("KeyOnlyHeader")` as a shorthand.

Reply:
Sounds good, just wanted to check